Fetch documents with a headless browser - refactored and updated (#189)
ambanum/CGUs: CONTRIBUTING.md

  "documents": {
    "<document type>": {
      "fetch": "<the URL where the document can be found>",

>       "executeClientScripts": "<a boolean which allow client scripts to be executed before accessing the content>",

Reviewer: ```suggestion
      "executeClientScripts": "<a boolean to execute client-side JavaScript loaded by the document before accessing the content, in case the DOM modifications are needed to access the content; defaults to false (fetch HTML only)>",
```

> We intend to expand coverage, but we focus for the moment on this subset of documents to fine-tune the system.

### `executeClientScripts`

> This boolean property, when set to `true` executes client script before trying to get document content by browsing to the URL with an headless browser. It's set to `false` by default. It is useful when the content of the document is loaded or modified dynamically on the browser side by client scripts.

Reviewer: ```suggestion
In some cases, the content of the document is only loaded (or is modified dynamically) by client scripts.
When set to `true`, this boolean property loads the page in a headless browser to load all assets and execute client scripts before trying to get document content.
Since the performance cost of this approach is high, it is set to `false` by default, relying on the HTML content only.
```

> **:warning: Browsing with an headless browser is expensive and time consuming, use only when necessary.**

Reviewer: I'm not sure this is needed after the rewrite above. In case it is worth adding, see suggestion below 🙂 

```suggestion
**:warning: Fetching content with a headless browser is expensive and time consuming, use only when necessary.**
```
